src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions_SharedWithCodeStyle.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery
+{
+    internal static partial class SyntaxTokenExtensions
+    {
+        public static bool IsLastTokenOfNode<T>(this SyntaxToken token)
+            where T : SyntaxNode
+        {
+            var node = token.GetAncestor<T>();
+            return node != null && token == node.GetLastToken(includeZeroWidth: true);
+        }
+    }
```

Comment:
probably far too much effort.  but we could consider a single PR that just makes the partials and pulls them out across the boundaries.  then, followup PRs would be more of hte interesting meat, and less of hte mechanical extraction.

Reply:
I'm happy to split the extensions files in a separate pull request if that will notably simplify this change.

Reply:
Up to you!  This PR was thankfully small enough that that part turned out to not be problematic.